### PR TITLE
Fix: Aliases Of Bidders With UserSync Supports Declared Only

### PR DESCRIPTION
**Pull request by SyntaxNode**

### CODE REVIEW

File: config/bidderinfo_test.go [hunk 1]
```diff
@@ -1522,6 +1552,75 @@ func TestSyncerEndpointOverride(t *testing.T) {
...
+			expected:    true,
+		},
+		{
+			description: "iframe-only",
+			givenSyncer: &Syncer{IFrame: &SyncerEndpoint{}},
+			expected:    true,
+		},
+		{
+			description: "redirect-only",
+			givenSyncer: &Syncer{IFrame: &SyncerEndpoint{}},
```
Comment (bsardo) [inline config/bidderinfo_test.go:1583]: `IFrame` should be `Redirect`

File: config/bidderinfo_test.go [hunk 2]
```diff
@@ -1522,6 +1552,75 @@ func TestSyncerEndpointOverride(t *testing.T) {
...
+		{
+			description: "supports-with-other",
+			givenSyncer: &Syncer{Key: "anyKey", Supports: []string{"anySupports"}},
+			expected:    true,
+		},
+	}
+
+	for _, test := range testCases {
+		result := test.givenSyncer.Defined()
+		assert.Equal(t, test.expected, result, test.description)
```
Comment (bsardo) [inline config/bidderinfo_test.go:1620]: Use `t.Run`?

**Review by bsardo — APPROVED** [review]

File: config/bidderinfo.go [hunk 3]
```diff
@@ -201,6 +201,21 @@ func (bi BidderInfo) IsEnabled() bool {
 	return !bi.Disabled
 }
 
+// Defined returns true if at least one field exists, except for the supports field.
+func (s *Syncer) Defined() bool {
+	if s == nil {
+		return false
+	}
+
+	return s.Key != "" ||
```
Comment (hhhjort) [inline config/bidderinfo.go:210]: For the purposes of this check, do we care only that at least one field has a non-default value, or do we want to verify that there is enough to actually use? For example, we need one of IFrame and Redirect to be non-nil. We can't do much if everything except SupportCORS is nil.

Comment (bsardo) [reply config/bidderinfo.go:210]: Hm good question. I think the presence of any of these fields implies that there is intent to define a user sync. If it is partially defined but not good enough to actually use I suppose we should create the user sync and expect the failure on startup. That's how you appear to have coded it @SyntaxNode. Is that your take?

**Review by hhhjort — APPROVED** [review]
LGTM

File: config/bidderinfo.go (hunk 3 above)
Comment (SyntaxNode) [reply config/bidderinfo.go:210]: Correct. This is our best attempt to determine if the human writing the config for the bidder intends for the usersync to work.

**Pull request merged**